modules/statistics/php/stats_demographic.class.inc

```diff
@@ -55,38 +56,38 @@ class Stats_Demographic extends \NDB_Form
     /**
      * RenderStatsTable function
      *
-     * @param string $sectionHeader     the value of sectionHeader
-     * @param string $tableHeader       the value of tableHeader
-     * @param string $subcats           the value of subcats
-     * @param string $visits            the value of visits
-     * @param string $dropdown_name     the value of dropdown_name
-     * @param string $dropdown_opt      the value of dropdown_opt
-     * @param string $dropdown_selected the value of dropdown_selected
-     * @param string $centres           the value of centres
-     * @param string $data              the value of data
-     * @param string $Subsection        the value of Subsection=''
-     * @param string $disclamer         the value of disclamer=''
-     * @param string $projectID         the value of projectID is null
+     * @param string $sectionHeader    the value of sectionHeader
+     * @param string $tableHeader      the value of tableHeader
+     * @param string $subcats          the value of subcats
+     * @param string $visits           the value of visits
+     * @param string $dropdownName     the value of dropdownName
+     * @param string $dropdownOpt      the value of dropdownOpt
+     * @param string $dropdownSelected the value of dropdownSelected
+     * @param string $centers          the value of centers
+     * @param string $data             the value of data
+     * @param string $subsection       the value of subsection=''
+     * @param string $disclamer        the value of disclamer=''
+     * @param string $projectID        the value of projectID is null
      *
      * @return string
      */
     function renderStatsTable($sectionHeader,
         $tableHeader,
         $subcats,
         $visits,
-        $dropdown_name,
-        $dropdown_opt,
-        $dropdown_selected,
-        $centres,
+        $dropdownName,
+        $dropdownOpt,
+        $dropdownSelected,
+        $centers,
         $data,
-        $Subsection="",
+        $subsection="",
         $disclamer='',
         $projectID=null
     ) {
-
-        $tpl_data = array();
+        $projectID = empty($projectID) ? null : $projectID;
```

Comment:
I'm pretty sure you could use the null coalesce operator here: 
`$projectID = $projectID ?? null;`

Reply:
@HenriRabalais I thought so too at first but it's more subtle than that: this line says "when `$projectID` is empty, instead make it null. Otherwise leave it alone."

Null coalesce would not make this change as it only kicks in when a variable is not set (different from when it IS set to empty).

Reply:
This line could probably benefit from a comment in the code actually.

Reply:
Makes sense!

Reply:
@zaliqarosli This is what that line is doing.

@ZainVirani This line definitely needs a comment. That's the 3rd person who has been tripped up by it.